services/authoring/src/repositories/git-repository.ts

```diff
@@ -179,17 +184,27 @@ export default class GitRepository {
   }
 
   async _pushRepositoryChanges(actionName) {
-    console.log('pushing changes:', actionName);
-
-    const remote = await this._repo.getRemote('origin', undefined);
-    await remote.push(['refs/heads/master:refs/heads/master'], <any>this._operationSettings, undefined);
+    try {
+      await new Promise((resolve, reject) => this._simpleRepo.push('origin', 'master', (err) => {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      }));
+    }
+    catch (ex) {
+      if (ex.includes("400 Bad Request")) {
```

Comment:
is there a better indication like a status code that we can use instead of parsing the message?

Reply:
Not really, it's just part of the stdout/stderr of git cli